tests/cpp/test_learner.cc

```diff
@@ -115,4 +116,118 @@ TEST(Learner, SLOW_CheckMultiBatch) {
   learner->UpdateOneIter(0, dmat.get());
 }
 
+#if defined(XGBOOST_USE_CUDA)
+
+TEST(Learner, IO) {
+  using Arg = std::pair<std::string, std::string>;
+  size_t constexpr kRows = 10;
+  auto pp_dmat = CreateDMatrix(kRows, 10, 0);
+  auto p_dmat = *pp_dmat;
+
+  std::vector<bst_float> labels(kRows);
+  for (size_t i = 0; i < labels.size(); ++i) {
+    labels[i] = i;
+  }
+  p_dmat->Info().labels_.HostVector() = labels;
+  std::vector<std::shared_ptr<DMatrix>> mat {p_dmat};
+
+  std::unique_ptr<Learner> learner {Learner::Create(mat)};
+  learner->Configure({Arg{"predictor", "gpu_predictor"},
```

Comment:
I'm assuming `tree_method` is set to `auto` (CPU algorithm) here? Let's explicitly write it out here